web/packages/teleport/src/services/userEvent/types.ts

```diff
@@ -76,8 +81,71 @@ export enum IntegrationEnrollKind {
   MachineIDKubernetes = 'INTEGRATION_ENROLL_KIND_MACHINE_ID_KUBERNETES',
   EntraId = 'INTEGRATION_ENROLL_KIND_ENTRA_ID',
   DatadogIncidentManagement = 'INTEGRATION_ENROLL_KIND_DATADOG_INCIDENT_MANAGEMENT',
+  AwsIdentityCenter = 'INTEGRATION_ENROLL_KIND_AWS_IDENTITY_CENTER',
+}
+
+/**
+ * IntegrationEnrollStep defines configurable steps for an integration type.
+ */
+export enum IntegrationEnrollStep {
+  /**
+   * AWSIC steps defined for AWS Idenity Center plugin.
+   * Value matches with proto enums defined in the backend.
+   */
+  ConnectOidc = 'INTEGRATION_ENROLL_STEP_AWSIC_CONNECT_OIDC',
+  ImportResourceSetDefaultOwner = 'INTEGRATION_ENROLL_STEP_AWSIC_SET_ACCESSLIST_DEFAULT_OWNER',
+  IdentitySourceUploadSamlMetadata = 'INTEGRATION_ENROLL_STEP_AWSIC_UPLOAD_AWS_SAML_SP_METADATA',
+  ScimTestConnection = 'INTEGRATION_ENROLL_STEP_AWSIC_TEST_SCIM_CONNECTION',
```

Comment:
Wouldn't you want these to match enum values instead, as in integers? I assume the web API is going to return those integers, not string values, maybe that assumption is wrong.

Reply:
I did such integer definition when I previously added events but after working in the discover components, I appreciated the shorter name by defining it this way `IntegrationEnrollStep.ConnectOidc` vs `IntegrationEnrollStep.INTEGRATION_ENROLL_STEP_AWSIC_CONNECT_OIDC` . The conversion to the integer happens in the web api where we convert it to the proto type.

Though the best way to do is to definitely defining exact `1:1` enum name and value. If this was auto generated `*.pb.ts`, I would have lived with the longer name but hand defining them lets us do some conversion in the backend and save us from looong name.

Reply:
Makes sense, thanks for giving some background on this.